packages/ddp/livedata_connection.js

```diff
@@ -520,12 +532,41 @@ _.extend(Connection.prototype, {
         if (!existing.ready)
           existing.readyCallback = callbacks.onReady;
       }
+
+      // XXX COMPAT WITH 1.0.3.1 we used to have onError but now we call
+      // onStopped with an optional error argument
       if (callbacks.onError) {
         // Replace existing callback if any, so that errors aren't
         // double-reported.
         existing.errorCallback = callbacks.onError;
       }
+
+      if (callbacks.onStopped) {
+        existing.stoppedCallback = callbacks.onStopped;
+      }
+
+      // Return the exact same handle so that we can use === to check if the
```

Comment:
In order to stop all of the subscriptions when the template is destroyed, I need to keep a list of all of the handles. To stop that list from growing without bound, I need to be able to deduplicate them. My two options were either using `===` to compare handles or having the handle include the subscription id. I thought not leaking the ID was the less risky option.